framework/include/linearfvkernels/LinearFVElementalKernel.h

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "LinearFVKernel.h"
+#include "ElemInfo.h"
+
+/**
+ * Finite volume kernel that contributes approximations of volumetric integral terms to the matrix
+ * and right hand side of a linear system.
+ */
+class LinearFVElementalKernel : public LinearFVKernel
+{
+public:
+  static InputParameters validParams();
+
+  /**
+   * Class constructor.
+   * @param parameters The InputParameters for the object
+   */
+  LinearFVElementalKernel(const InputParameters & params);
+
+  virtual void addMatrixContribution() override;
+
+  virtual void addRightHandSideContribution() override;
+
+  /**
+   * Set the current ElemInfo object.
+   * @param elem_info The new ElemInfo which will be used as the current
+   */
+  void setCurrentElemInfo(const ElemInfo * elem_info) { _current_elem_info = elem_info; }
+
+  /// Computes the system matrix contribution for the current element
```

Comment:
and the current variable right?
since there's no AD are devs supposed to implement coupled terms there too?

Reply:
Yes. But we need to implement coupled terms in the monolithic AD version too.